src/Server/StaticRequestHandler.cpp

```diff
@@ -101,7 +101,13 @@ static inline void trySendExceptionToClient(
 
 void StaticRequestHandler::handleRequest(HTTPServerRequest & request, HTTPServerResponse & response)
 {
-    auto keep_alive_timeout = server.config().getUInt("keep_alive_timeout", 10);
+    const auto & config = server.config();
+    auto keep_alive_timeout = config.getUInt("keep_alive_timeout", 10);
+    size_t hsts_max_age = config.getUInt64("hsts_max_age", 0);
+
+    if (hsts_max_age > 0)
+        response.add("Strict-Transport-Security", "max-age=" + std::to_string(hsts_max_age));
```

Comment:
It should not be added for HTTP requests.
And it should also be present for query responses over HTTPS, not only for static responses.

Reply:
@alexey-milovidov updated

Reply:
updated